spec/models/instance_spec.rb

```diff
@@ -1,6 +1,29 @@
 require 'rails_helper'
 
 RSpec.describe Instance, type: :model do
+  let(:instance) { build(:instance) }
+
+  context 'when host format is invalid' do
+    it 'is expected to be invalid' do
+      hosts = ['example,com', 'www_.example.org', 'example_.org',
+               'example.org_', 'example..org', 'example.org.']
+      hosts.each do |invalid_host|
+        instance.host = invalid_host
+        expect(instance).not_to be_valid
+      end
+    end
+  end
+
+  context 'when host format is valid' do
+    it 'is expected to be valid' do
+      hosts = ['example.ORG', 'www.example.org', 'ex-ample.org', 'example.org.sg']
```

Comment:
Actually I tried, but that one is not possible, we cannot nested one test case inside a loop.

http://stackoverflow.com/questions/16698396/need-to-loop-through-an-array-in-rspec-test-not-running